src/client/BuilderClient.ts

```diff
@@ -1,50 +1,34 @@
-import axios, { AxiosInstance } from 'axios'
 import FormData from 'form-data'
 import { Authenticator, AuthIdentity } from 'dcl-crypto'
 import { Content } from '../content/types'
 import { RemoteItem, LocalItem } from '../item/types'
 import { ClientError } from './BuilderClient.errors'
 import { ServerResponse } from './types'
+import crossFetch from 'cross-fetch'
 
 export class BuilderClient {
-  private axios: AxiosInstance
+  private fetch: (url: string, init?: RequestInit) => Promise<Response>
   private readonly AUTH_CHAIN_HEADER_PREFIX = 'x-identity-auth-chain-'
 
   constructor(
     url: string,
     private identity: AuthIdentity,
-    private address: string
+    private address: string,
+    externalFetch: typeof fetch = crossFetch
   ) {
-    this.axios = axios.create({
-      baseURL: url
-    })
-
-    // Uses the interceptors API to add the auth headers to all requests
-    this.axios.interceptors.request.use((config) => {
-      if (!config.method || !config.url) {
-        return config
-      }
-
-      const headers = this.createAuthHeaders(
-        config.method,
-        config.url.replace(/\/v[0-9]/, '')
-      )
-      config.headers = { ...config.headers, ...headers }
-
-      return config
-    })
+    this.fetch = (...args) => {
+      const path = args[0]
+      const method: string = args[1]?.method ?? args[0] ?? 'get'
```

Comment:
this is completely fine, but if we know that we're getting the first and second arguments and we're not re-spreading them into another function like this:

```ts
funct(...args) { 
  otherFunc(...args)
}
```

why not just getting 

```ts
this.fetch = (path: string, req: Request)
```

(I have no idea what the second param is...and that's the point :P)

Reply:
You're right, there's no reason to use args there! I left that because I was going to use the fetch method exactly as the one from the dom but I had to change it.